### PR TITLE
Additional synapse.lib.net tests and bugfixes

### PR DESCRIPTION
**Pull request by vertexmc**

### CODE REVIEW

File: synapse/neuron.py [hunk 1]
```diff
@@ -490,11 +491,10 @@ def onlink(ok, link):
 
             isok, sess = retn.wait(timeout=timeout)
             if not isok:
-                # XXX Raise / log here?
-                return None
+                raise s_common.RetnTimeout(msg='retnwait timed out or failed')
```
Comment (vEpiphyte) [inline synapse/neuron.py]: if isok is false, there will be a excfo tufo in ``sess`` so raising RetnTimeout is incorrect (there could be something else there)

File: synapse/neuron.py [hunk 2]
```diff
@@ -490,11 +491,10 @@ def onlink(ok, link):
...
 
             if not sess.waittx(timeout=timeout):
-                return None
+                raise s_common.RetnTimeout(msg='waittx timed out or failed')
```
Comment (vEpiphyte) [inline synapse/neuron.py]: i actually think that at this point we could leave the context manager (fini' the retnwait object) then do the sess.waittx(). Whether we raise or return None here is a different question; but raising RetnTimeout here doesn't make sense.

File: synapse/cryotank.py [hunk 3]
```diff
@@ -416,7 +416,7 @@ def puts(self, name, items, timeout=None):
                 return False
 
             iitr = s_common.chunks(items, self._chunksize)
-            return chan.txwind(iitr, 100, timeout=timeout)
+            return chan.txwind(iitr, self._chunksize, timeout=timeout)
```
Comment (vEpiphyte) [inline synapse/cryotank.py]: :+1:

File: synapse/neuron.py [hunk 4]
```diff
@@ -456,17 +456,19 @@ def __init__(self, auth, roots=()):
...
         '''
-        Open the Cell at the remote addr and return a UserSess Link.
+        Synchronously opens the Cell at the remote addr and return a UserSess Link.
 
         Args:
             addr ((str,int)): A (host, port) address tuple
             timeout (int/float): Connection timeout in seconds.
 
+        Raises:
+            CellUserErr: Raised if a timeout or link negotion fails.  May have
```
Comment (vEpiphyte) [inline synapse/neuron.py]: spelling

Comment (vertexmc) [reply synapse/neuron.py]: fixed

File: synapse/tests/test_lib_net.py [hunk 5]
```diff
@@ -54,3 +244,171 @@ def onlink(link):
...
+            self.isin(expected_msg, mesgs)
+
+    def test_lib_net_listenfail(self):
+        expected_msg = 'listen() onlink error'
+        with self.getLoggerStream('synapse.lib.net', expected_msg) as stream:
+
+            class LisnLink(s_net.Link):
+
+                def linked(self):
+                    steps.done('lisn')
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_net.py]: ``steps`` is not defined

File: synapse/tests/test_lib_net.py [hunk 6]
```diff
@@ -2,8 +2,198 @@
 
 from synapse.tests.common import *
 
+logger = logging.getLogger(__name__)
+
 class NetTest(SynTest):
 
+    def test_lib_net_chan_txfini(self):
+        class FiniableLink(s_net.Link):
+            def tx(msg, **kwargs):
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_net.py]: is ``self`` missing from this definition?

Comment (vertexmc) [reply synapse/tests/test_lib_net.py]: The first arg passed to the fn is the mesg. Looked weird to me as well

File: synapse/tests/test_lib_net.py (hunk 5 above)
Comment (vertexmc) [reply synapse/tests/test_lib_net.py]: This whole class can actually be removed

File: synapse/tests/test_lib_net.py [hunk 7]
```diff
@@ -2,8 +2,198 @@
...
+    def test_lib_net_link_onrx(self):
+        msg = 'woohoo'
+        def fn(self, *args, **kwargs):
+            logger.error('woohoo')
+
+        link = s_net.Link()
+        link.onrx(fn)
+        self.eq(link.rxfunc, fn)
+
+        with self.getLoggerStream('synapse.tests.test_lib_net', msg) as stream:
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_net.py]: in alot of cases with ``getLoggerStream`` you're providing the second parameter which will cause the streamevent to be set when the message occurs; but not always waiting for it. is there a reason for that? do most of the tests just run in mainthread?

File: synapse/tests/test_lib_net.py (hunk 5 above)
Comment (vertexmc) [reply synapse/tests/test_lib_net.py]: fixed

File: synapse/tests/test_lib_net.py (hunk 7 above)
Comment (vEpiphyte) [reply synapse/tests/test_lib_net.py]: also i would operate on the stream to read the messages outside of the ``with`` block to ensure that the stream has been removed from the logger and nothing more is written to it

Comment (vertexmc) [reply synapse/tests/test_lib_net.py]: No. I should fix this so that all the tests wait

File: synapse/tests/test_lib_net.py (hunk 6 above)
Comment (vEpiphyte) [reply synapse/tests/test_lib_net.py]:
```
diff --git a/synapse/tests/test_lib_net.py b/synapse/tests/test_lib_net.py
index c44a2f6..434b203 100644
--- a/synapse/tests/test_lib_net.py
+++ b/synapse/tests/test_lib_net.py
@@ -8,12 +8,12 @@ class NetTest(SynTest):
 
     def test_lib_net_chan_txfini(self):
         class FiniableLink(s_net.Link):
-            def tx(msg, **kwargs):
+            def tx(self, msg):
                 logger.error(msg)
 
         msg = '%r' % (('cool', {}),)
         with self.getLoggerStream('synapse.tests.test_lib_net', msg) as stream:
-            chan = s_net.Chan(FiniableLink, None)
+            chan = s_net.Chan(FiniableLink(), None)
             chan.txfini(data=('cool', {}))
             stream.seek(0)
             self.isin(msg, stream.read())
``` seems to work

File: synapse/tests/test_lib_net.py (hunk 7 above)
Comment (vertexmc) [reply synapse/tests/test_lib_net.py]: fixed

File: synapse/tests/test_lib_net.py (hunk 6 above)
Comment (vertexmc) [reply synapse/tests/test_lib_net.py]: this was wrong, fixed

**Pull request merged**